app/controllers/home_controller.rb

```diff
@@ -26,13 +26,7 @@ def show
   end
 
   def toggle_mobile
-    if session[:mobile_view].nil?
-      # we're most probably not on mobile, but user wants it anyway
-      session[:mobile_view] = true
-    else
-      # switch from mobile to normal html
-      session[:mobile_view] = !session[:mobile_view]
-    end
+    session[:mobile_view] = session[:mobile_view].nil ? true : !session[:mobile_view]
```

Comment:
Typo here, `.nil` -> `.nil?`. You can catch these easily by [running tests locally](https://wiki.diasporafoundation.org/Testing_workflow).